### PR TITLE
Expanded guide with testing assertion usage examples.

### PR DESCRIPTION
**Pull request by Hajto**

### CODE REVIEW

File: guide/creating_element/testing.md [hunk 1]
```diff
@@ -231,16 +231,79 @@ end
...
 caps to parent pipeline. Useful for asserting output buffers one by one.
 
+We also prepared a bunch of assertions that work in conjunction with both
+`Membrane.Testing.Pipeline` and said elements. They are stored in `Membrane.Testing.Assertions`,
+they usually come in two flavors: `assert_*` and `refute_*` similarly to `ExUnit` assertions.
+
+### Testing sink
+
+`Membrane.Testing.Sink` reports caps, events and buffers it receives to it's pipeline. If you are
+using it within `Membrane.Testing.Pipeline` you can make assertions on this data. For example
```
Comment (bblaszkow06) [inline guide/creating_element/testing.md]:
```
you can make assertions about what the `Membrane.Testing.Sink` have received
```

File: guide/creating_element/testing.md [hunk 2]
```diff
@@ -231,16 +231,79 @@ end
...
+# assert every message was received three times
+Enum.each(range, fn payload ->
+  assert_sink_buffer(pid, :sink1, %Buffer{payload: ^payload})
+  assert_sink_buffer(pid, :sink2, %Buffer{payload: ^payload})
+end)
+```
+
+### Testing element's interaction with Pipeline
+
+Other common use of these assertions is checking wether communication with Pipeline is proceeding
```
Comment (bblaszkow06) [inline guide/creating_element/testing.md]: **An**other

Comment (bblaszkow06) [inline guide/creating_element/testing.md]: w**h**eather

File: guide/creating_element/testing.md [hunk 3]
```diff
@@ -231,16 +231,79 @@ end
...
+correctly. You can either check wether Pipeline received a message that would be handled by
```
Comment (bblaszkow06) [inline guide/creating_element/testing.md]: w**h**eather

File: guide/creating_element/testing.md [hunk 4]
```diff
@@ -231,16 +231,79 @@ end
...
+assert_end_of_stream(pid, :sink2, :input, 3000)
+
+# assert every message was received three times
+Enum.each(range, fn payload ->
+  assert_sink_buffer(pid, :sink1, %Buffer{payload: ^payload})
+  assert_sink_buffer(pid, :sink2, %Buffer{payload: ^payload})
+end)
+```
+
+### Testing element's interaction with Pipeline
```
Comment (bblaszkow06) [inline guide/creating_element/testing.md]: interaction with `Membrane.Testing.Pipeline` or `a pipeline`

File: guide/creating_element/testing.md (hunk 2 above)
Comment (bblaszkow06) [inline guide/creating_element/testing.md]: change Pipeline like in header

File: guide/creating_element/testing.md [hunk 5]
```diff
@@ -231,16 +231,79 @@ end
...
+
+Other common use of these assertions is checking wether communication with Pipeline is proceeding
+correctly. You can either check wether Pipeline received a message that would be handled by
+`c:Membrane.Pipeline.handle_other/2`:
+
+```elixir
+assert_pipeline_receive(pipeline_pid, {:topic, _})
+```
+
+Such assertion would assert that message matching pattern `{:topic, _}` will be sent to
```
Comment (bblaszkow06) [inline guide/creating_element/testing.md]: Such an assertion

File: guide/creating_element/testing.md [hunk 6]
```diff
@@ -231,16 +231,79 @@ end
...
+pipeline with pid `pipeline_pid`.
+
+If you were for example to implement demuxer want to ensure pipeline is notified by demuxer asking
```
Comment (bblaszkow06) [inline guide/creating_element/testing.md]: add commas before and after `for example`

Comment (bblaszkow06) [inline guide/creating_element/testing.md]: implement a demuxer

Comment (bblaszkow06) [inline guide/creating_element/testing.md]: something is missing between demuxer and want

Comment (bblaszkow06) [inline guide/creating_element/testing.md]: by the demuxer

File: guide/creating_element/testing.md [hunk 7]
```diff
@@ -231,16 +231,79 @@ end
...
+for mapping of streams to pads you would do it like this:
```
Comment (bblaszkow06) [inline guide/creating_element/testing.md]: I would add comma after `pads`

File: guide/creating_element/testing.md [hunk 8]
```diff
@@ -231,16 +231,79 @@ end
 Apart from the `Membrane.Testing.Pipeline`, which we've already seen, there are a bunch of other
 testing utilities which may come in handy for different test scenarios:
 
-- `Membrane.Testing.Assertions` - Contains all available assertions that work with
```
[the remark this answers is not in the document]
Comment (Hajto) [reply guide/creating_element/testing.md]: This line does not contain `other testing utilities`, but I assumed you meant line 229 which contained `## using other...`

File: guide/creating_element/testing.md [hunk 9]
```diff
@@ -231,16 +231,79 @@ end
...
+        {:src, :output} => {:tee, :input},
+        {:tee, :output, 1} => {:sink1, :input},
+        {:tee, :output, 2} => {:sink2, :input}
+      }
+    })
+
+Membrane.Pipeline.play(pipeline)
+
+# Wait for EndOfStream message on every sink
+assert_end_of_stream(pid, :sink1, :input, 3000)
```
[the remark this answers is not in the document]
Comment (Hajto) [reply guide/creating_element/testing.md]: Should I also change this, here:
https://github.com/membraneframework/membrane-element-tee/blob/master/test/membrane_element_tee/parallel/pipeline_parallel_test.exs#L34 ??

File: guide/creating_element/testing.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (bblaszkow06) [reply guide/creating_element/testing.md]: Correct, it was out of scope of diff

File: guide/creating_element/testing.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (bblaszkow06) [reply guide/creating_element/testing.md]: I would appreciate that ;)

File: guide/creating_element/testing.md [hunk 10]
```diff
@@ -269,32 +269,32 @@ range = 1..100
...
 ```
 
-### Testing element's interaction with Pipeline
+### Testing element's interaction with a pipeline
 
-Other common use of these assertions is checking wether communication with Pipeline is proceeding
-correctly. You can either check wether Pipeline received a message that would be handled by
-`c:Membrane.Pipeline.handle_other/2`:
+Another common use of these assertions is checking weather communication with a pipeline is
+proceeding correctly. You can either check weather Pipeline received a message that would be
```
[the remark this answers is not in the document]
Comment (bblaszkow06) [reply guide/creating_element/testing.md]: Unless you meant the sunny weather outside, of course.

File: guide/creating_element/testing.md [hunk 11]
```diff
@@ -269,32 +269,32 @@ range = 1..100
...
+assert_end_of_stream(pid, :sink2, :input, 3000)
 ```
 
-### Testing element's interaction with Pipeline
+### Testing element's interaction with a pipeline
 
-Other common use of these assertions is checking wether communication with Pipeline is proceeding
-correctly. You can either check wether Pipeline received a message that would be handled by
-`c:Membrane.Pipeline.handle_other/2`:
+Another common use of these assertions is checking weather communication with a pipeline is
```
Comment (bblaszkow06) [inline guide/creating_element/testing.md]: whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether whether

File: guide/creating_element/testing.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (Hajto) [reply guide/creating_element/testing.md]: sorry

**Pull request merged**